components/style/properties/longhand/box.mako.rs

```diff
@@ -1470,48 +1420,40 @@ ${helpers.predefined_type("scroll-snap-coordinate",
                         let ty = try!(specified::LengthOrPercentage::parse(context, input));
                         try!(input.expect_comma());
                         let tz = try!(specified::Length::parse(context, input));
-                        result.push(SpecifiedOperation::Translate(
-                            TranslateKind::Translate3D,
-                            tx,
-                            ty,
-                            tz));
+                        result.push(SpecifiedOperation::Translate3D(tx, ty, tz));
                         Ok(())
                     }))
-
                 },
                 "scale" => {
                     try!(input.parse_nested_block(|input| {
-                        let (sx, sy) = try!(parse_two_numbers(input));
-                        result.push(SpecifiedOperation::Scale(sx,
-                                                              sy,
-                                                              Number::new(1.0)));
+                        let sx = try!(specified::parse_number(input));
+                        if let Ok(()) = input.expect_comma() {
```

Comment:
Also, let's use just `if input.try(|input| input.expect_comma()).is_ok()` instead of `if let`, for consistency with all the other existing code.